roles/dns_encryption/tasks/arch.yml

```diff
@@ -0,0 +1,13 @@
+---
+
+- name: Arch Linux | Install dnscrypt-proxy
+  pacman:
+    name: dnscrypt-proxy
+    state: latest
+
+- name: Arch Linux | Patch dnscrypt-proxy.service
+  template:
+    src: dnscrypt-proxy.service.j2
+    dest: "{{ config_prefix|default('/') }}etc/systemd/system/dnscrypt-proxy.service"
```

Comment:
I don't like this part very much.

Here I patch the service file to disable SystemD socket activation which Arch package provides by default.

I did it because I coulnd't make `dnscrypt-proxy.socket` start **after** activating `172.16.0.1` loopback interface. Though if I maneged to do it I still had to patch `dnscrypt-proxy.socket` to change the listen address.

Reply:
I am not sure if a dependency on `dnscrypt-proxy.socket` can be removed this way.